contracts/strategies/JoeStrategyV1.sol

```diff
@@ -0,0 +1,365 @@
+     * @param maxAmountIn1 amount token1
+     * @return liquidity tokens
+     */
+    function _addLiquidity(address token0, address token1, uint maxAmountIn0, uint maxAmountIn1) private returns (uint) {
+        (uint112 reserve0, uint112 reserve1,) = IPair(address(depositToken)).getReserves();
+        uint amountIn1 = _quoteLiquidityAmountOut(maxAmountIn0, reserve0, reserve1);
+        if (amountIn1 > maxAmountIn1) {
+            amountIn1 = maxAmountIn1;
+            maxAmountIn0 = _quoteLiquidityAmountOut(maxAmountIn1, reserve1, reserve0);
+        }
+        
+        _safeTransfer(token0, address(depositToken), maxAmountIn0);
+        _safeTransfer(token1, address(depositToken), amountIn1);
+        return IPair(address(depositToken)).mint(address(this));
+    }
+
+    /**
+     * @notice Swap directly through a Pair
+     * @param amountIn input amount
+     * @param fromToken address
+     * @param toToken address
+     * @param pair Pair used for swap
+     * @return output amount
+     */
+    function _swap(uint amountIn, address fromToken, address toToken, IPair pair) private returns (uint) {
+        (address token0,) = _sortTokens(fromToken, toToken);
+        (uint112 reserve0, uint112 reserve1,) = pair.getReserves();
+        if (token0 != fromToken) (reserve0, reserve1) = (reserve1, reserve0);
+        uint amountOut1 = 0;
+        uint amountOut2 = _getAmountOut(amountIn, reserve0, reserve1);
+        if (token0 != fromToken) (amountOut1, amountOut2) = (amountOut2, amountOut1);
+        _safeTransfer(fromToken, address(pair), amountIn);
+        pair.swap(amountOut1, amountOut2, address(this), zeroBytes);
+        return amountOut2 > amountOut1 ? amountOut2 : amountOut1;
+    }
+
+
+    /**
+     * @notice Converts reward tokens to deposit tokens
+     * @dev No price checks enforced
+     * @param amount reward tokens
+     * @return deposit tokens
+     */
+    function _convertWAVAXToDepositTokens(uint amount) private returns (uint) {
+        uint amountIn = amount.div(2);
+        require(amountIn > 0, "JoeStrategyV1::_convertRewardTokensToDepositTokens");
+
+        address token0 = IPair(address(depositToken)).token0();
+        uint amountOutToken0 = amountIn;
+        if (address(WAVAX) != token0) {
+            amountOutToken0 = _swap(amountIn, address(WAVAX), token0, swapPairToken0);
+        }
+
+        address token1 = IPair(address(depositToken)).token1();
+        uint amountOutToken1 = amountIn;
+        if (address(WAVAX) != token1) {
+            amountOutToken1 = _swap(amountIn, address(WAVAX), token1, swapPairToken1);
+        }
+
+        return _addLiquidity(token0, token1, amountOutToken0, amountOutToken1);
+    }
+    
+    function checkReward() public override view returns (uint) {
+        (uint pendingJoe, address extraRewardToken, , uint pendingExtraToken) = stakingContract.pendingTokens(PID, address(this));
+        if (extraRewardToken == address(rewardToken)){
```

Comment:
We could ignore this extra token -- it should definitely not get added to Joe, as the exchange rates are diff

Reply:
simplifiest indeed